ir/src/main/scala/org/scalajs/core/ir/Tags.scala

```diff
@@ -90,6 +90,7 @@ private[ir] object Tags {
 
   // TODO Reorganize these when we can break binary compatibility
   final val TagJSSpread = TagModuleExportDef + 1
+  final val TagRestParamDef = TagJSSpread + 1
```

Comment:
Why are you adding a separate tag for this? So far we had a tag per case class... You can easily support the same tag by changing the deserialization behavior based on the version...